scripts/validate-front-matter/script.py

```diff
@@ -0,0 +1,303 @@
+import datetime
+import os
+import re
+import sys
+
+from colored import fg, bg, attr
+
+from yaml import load
+try:
+    from yaml import CLoader as Loader
+except ImportError:
+    from yaml import Loader
+
+path = 'src/content'
+changes_path = 'src/content/changes/'
+crds_path = 'src/content/ui-api/management-api/crd/'
+
+INVALID_LAST_REVIEW_DATE = 'INVALID_LAST_REVIEW_DATE'
+INVALID_OWNER            = 'INVALID_OWNER'
+LONG_DESCRIPTION         = 'LONG_DESCRIPTION'
+LONG_LINK_TITLE          = 'LONG_LINK_TITLE'
+LONG_TITLE               = 'LONG_TITLE'
+LONG_USER_QUESTIONS      = 'LONG_USER_QUESTIONS'
+NO_DESCRIPTION           = 'NO_DESCRIPTION'
+NO_FRONT_MATTER          = 'NO_FRONT_MATTER'
+NO_LAST_REVIEW_DATE      = 'NO_LAST_REVIEW_DATE'
+NO_LINK_TITLE            = 'NO_LINK_TITLE'
+NO_OWNER                 = 'NO_OWNER'
+NO_QUESTION_MARK         = 'NO_QUESTION_MARK'
+NO_TITLE                 = 'NO_TITLE'
+NO_TRAILING_NEWLINE      = 'NO_TRAILING_NEWLINE'
+NO_USER_QUESTIONS        = 'NO_USER_QUESTIONS'
+NO_WEIGHT                = 'NO_WEIGHT'
+REVIEW_TOO_LONG_AGO      = 'REVIEW_TOO_LONG_AGO'
+SHORT_DESCRIPTION        = 'SHORT_DESCRIPTION'
+SHORT_TITLE              = 'SHORT_TITLE'
+UNKNOWN_ATTRIBUTE        = 'UNKNOWN_ATTRIBUTE'
+
+# All checks with metadata, in a logical order
+checks = (
```

Comment:
If it is planned to expand the validator with more checks in the future, I would suggest the following:

- Make a data structure which assigns front matter keys to checks which apply to it
- make checks functions
- Iterate over the found keys in the frontmatter and execute the check functions for them
-

Reply:
Sounds good!

Reply:
(Keeping this for the future indeed)